lib/route/link.c

```diff
@@ -2273,6 +2284,17 @@ uint8_t rtnl_link_get_carrier(struct rtnl_link *link)
 }
 
 /**
+ * Return carrier on/off changes of link object
+ * @arg link		Link object
+ *
+ * @return Carrier changes.
+ */
+uint32_t rtnl_link_get_carrier_changes(struct rtnl_link *link)
```

Comment:
It's not clear what's best, and consistency should be aimed for... but if the return value is uint32_t, there is no space to encode an error. Or even to encode whether the attribute is present -- which might be relevant information. I think a `int` error code is better.